Modules/StudyProgramme/classes/class.ilObjStudyProgrammeSettingsGUI.php

```diff
@@ -166,21 +168,22 @@ protected function update() {
 		// This could further improved by providing a new container for asynch-forms in the
 		// UI-Framework.
 		$update_possible = !is_null($content);
-		if ($update_possible) {
-			$this->updateWith($prg, $content);
-			ilUtil::sendSuccess($this->lng->txt("msg_obj_modified"),true);
-			$response = ilAsyncOutputHandler::encodeAsyncResponse(array("success"=>true, "message"=>$this->lng->txt("msg_obj_modified")));
-		} else {
-			ilUtil::sendFailure($this->lng->txt("msg_form_save_error"));
-			$response = ilAsyncOutputHandler::encodeAsyncResponse(array("success"=>false, "errors"=>$form->getErrors()));
-		}
-
 		if($this->ctrl->isAsynch()) {
+			if ($update_possible) {
+				$this->updateWith($prg, $content);
```

Comment:
This needs to be executed in the non-asynch-case as well. I would suggest to leave it where it is.

Reply:
@klees I changed this.